sequencer/finalizer.go

```diff
@@ -13,38 +19,59 @@ type finalizer struct {
 	GERCh                chan common.Hash
 	L2ReorgCh            chan struct{}
 	SendingToL1TimeoutCh chan bool
-	SequencerAddress     common.Address
+	sequencerAddress     common.Address
 	worker               workerInterface
 	dbManager            dbManagerInterface
-	executor             executorInterface
+	executor             stateInterface
+	maxTxsPerBatch       uint64
+}
+
+func newFinalizer(worker workerInterface, dbManager dbManagerInterface, executor stateInterface, sequencerAddr common.Address, maxTxsPerBatch uint64) *finalizer {
+	return &finalizer{
+		ForcedBatchCh:        make(chan state.Batch),
+		GERCh:                make(chan common.Hash),
+		L2ReorgCh:            make(chan struct{}),
+		SendingToL1TimeoutCh: make(chan bool),
+		sequencerAddress:     sequencerAddr,
+		worker:               worker,
+		dbManager:            dbManager,
+		executor:             executor,
+		maxTxsPerBatch:       maxTxsPerBatch,
+	}
 }
 
 type wipBatch struct {
 	batchNumber           uint64
 	initialStateRoot      common.Hash
 	intermediaryStateRoot common.Hash
 	timestamp             uint64
-	GER                   common.Hash // 0x000...0 means to not update
+	GER                   common.Hash // 0x000...0 (ZeroHash) means to not update
 	txs                   []TxTracker
-	remainingResources    RemainingResources // Decide if we use remaining or used to keep track of the current batch resources
+	remainingResources    RemainingResources
 }
 
-func newFinalizer(worker workerInterface, dbManager dbManagerInterface, executor executorInterface) *finalizer {
-	// TODO: Init channels
-	return &finalizer{worker: worker, dbManager: dbManager, executor: executor}
-}
-
-func (f *finalizer) Start() {
-	// TODO: Load last batch from DB
+func (f *finalizer) Start(ctx context.Context) {
+	lastBatch, err := f.dbManager.GetLastBatch(ctx)
+	if err != nil {
+		log.Fatal("failed to get last batch")
+	}
 
-	wipBatch := wipBatch{}
+	wipBatch := wipBatch{
+		GER:                   lastBatch.GlobalExitRoot,
+		initialStateRoot:      lastBatch.StateRoot,
+		intermediaryStateRoot: lastBatch.StateRoot,
```

Comment:
This state root should be taken from the latest tx included into the batch. You can do this in a later PR as here the priority is to define interfaces ASAP

Reply:
This I had in mind just wanted to make quick PR for the interfaces.